app/src/main/java/com/framgia/soundclound/data/source/local/realm/TrackLocalDataSourceimpl.java

```diff
@@ -0,0 +1,92 @@
+package com.framgia.soundclound.data.source.local.realm;
+
+import android.content.Context;
+
+import com.framgia.soundclound.data.model.Track;
+import com.framgia.soundclound.data.source.TrackDataSource;
+import com.framgia.soundclound.data.source.local.SharePreferences;
+import com.framgia.soundclound.util.Constant;
+
+import java.util.List;
+
+import io.realm.RealmResults;
+
+/**
+ * Created by ADMIN on 1/14/2018.
+ */
+
+public class TrackLocalDataSourceimpl implements TrackDataSource.TrackLocalDataSoure {
+
+    private RealmApi mRealmApi;
+    private static TrackLocalDataSourceimpl sTrackLocalDataSourceimpl;
```

Comment:
WARNING: [StaticFieldLeak] Do not place Android context classes in static fields (static reference to `TrackLocalDataSourceimpl` which has field `mContext` pointing to `Context`); this is a memory leak (and also breaks Instant Run)
 A static field will leak contexts.

Non-static inner classes have an implicit reference to their outer class. If that outer class is for example a `Fragment` or `Activity`, then this reference means that the long-running handler/loader/task will hold a reference to the activity which prevents it from getting garbage collected.

Similarly, direct field references to activities and fragments from these longer running instances can cause leaks.

ViewModel classes should never point to Views or non-application Contexts.